container/state.go

```diff
@@ -271,3 +282,27 @@ func (s *State) SetDead() {
 	s.Dead = true
 	s.Unlock()
 }
+
+// SetCheckpointed sets the container's status to indicate it has been checkpointed
+func (s *State) SetCheckpointed(leaveRunning bool) {
+	s.Lock()
+	s.CheckpointedAt = time.Now().UTC()
+	s.Checkpointed = !leaveRunning
+	s.Running = leaveRunning
+	s.Paused = false
+	s.Restarting = false
+	// FIXME: Not sure if we need to close and recreate waitChan.
+	// close(s.waitChan)
+	// s.waitChan = make(chan struct{})
+	s.Unlock()
+}
+
+// HasBeenCheckpointed indicates whether the container has ever been checkpointed
+func (s *State) HasBeenCheckpointed() bool {
+	return s.CheckpointedAt != time.Time{}
```

Comment:
I think other docker code is using `IsZero` instead of comparing to the emtpy Time struct